tests/rule_based_profiler/test_profiler_user_workflows.py

```diff
@@ -109,75 +205,200 @@ def test_alice_columnar_table_single_batch_batches_are_accessible(
+    datasource: Datasource = cast(Datasource, context.datasources[datasource_name])
+    data_connector: DataConnector = datasource.data_connectors[data_connector_name]
 
-    file_list = [alice_columnar_table_single_batch["sample_data_relative_path"]]
+    file_list: List[str] = [
+        alice_columnar_table_single_batch["sample_data_relative_path"]
+    ]
 
     assert (
         data_connector._get_data_reference_list_from_cache_by_data_asset_name(
-            data_asset_name=asset_name
+            data_asset_name=data_asset_name
         )
         == file_list
     )
 
-    batch_request_1 = BatchRequest(
+    batch_request_1: BatchRequest = BatchRequest(
         datasource_name=datasource_name,
         data_connector_name=data_connector_name,
-        data_asset_name=asset_name,
+        data_asset_name=data_asset_name,
         data_connector_query={
             "index": -1,
         },
     )
     # Should give most recent batch
-    validator_1 = context.get_validator(
+    validator_1: Validator = context.get_validator(
         batch_request=batch_request_1,
         create_expectation_suite_with_name="my_expectation_suite_name_1",
     )
-    metric_max = validator_1.get_metric(
+    metric_max: int = validator_1.get_metric(
         MetricConfiguration("column.max", metric_domain_kwargs={"column": "event_type"})
     )
     assert metric_max == 73
 
 
 def test_alice_profiler_user_workflow_single_batch(
-    alice_columnar_table_single_batch_context, alice_columnar_table_single_batch
+    alice_columnar_table_single_batch_context,
+    alice_columnar_table_single_batch,
 ):
     # Load data context
     data_context: DataContext = alice_columnar_table_single_batch_context
     # Load profiler configs & loop (run tests for each one)
     profiler_config: str = alice_columnar_table_single_batch["profiler_config"]
 
     # Instantiate Profiler
-    full_profiler_config_dict = yaml.load(profiler_config)
-    rules_configs = full_profiler_config_dict.get("rules")
-    variables_configs = full_profiler_config_dict.get("variables")
-    profiler = Profiler(
+    full_profiler_config_dict: dict = yaml.load(profiler_config)
+    rules_configs: dict = full_profiler_config_dict.get("rules")
+    variables_configs: dict = full_profiler_config_dict.get("variables")
+
+    datasource_name: str = "alice_columnar_table_single_batch_datasource"
+    data_connector_name: str = "alice_columnar_table_single_batch_data_connector"
+    data_asset_name: str = "alice_columnar_table_single_batch_data_asset"
+
+    validator: Validator = data_context.get_validator(
+        datasource_name=datasource_name,
+        data_connector_name=data_connector_name,
+        data_asset_name=data_asset_name,
+        create_expectation_suite_with_name=alice_columnar_table_single_batch[
+            "expected_expectation_suite_name"
+        ],
+    )
+
+    profiler: Profiler = Profiler(
+        validator=validator,
```

Comment:
What changed here? Did we not previously need to pass a `validator` into a `Profiler`? How did that work?

Reply:
@talagluck This was probably just the moving of code (if GitHub shows the plus signs).